# 1.全体の解釈

In [3]:
import pickle

path = './model.pickle'

with open(path, 'rb') as f:
    analysis = pickle.load(f)

analysis.summary()


<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                    魅力度   R-squared:                       0.595
Model:                            OLS   Adj. R-squared:                  0.567
Method:                 Least Squares   F-statistic:                     21.04
Date:                Tue, 12 Nov 2024   Prob (F-statistic):           1.53e-08
Time:                        11:39:36   Log-Likelihood:                -163.04
No. Observations:                  47   AIC:                             334.1
Df Residuals:                      43   BIC:                             341.5
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const         18.6402      2.117      8.806      0.000      14.371      22.909
都道府県       -2.828e-09   3.21e-10     -8.806      0.000   -3.48e-09   -2.18e-09
人口         -2.304e-06   7.99e-07     -2.883      0.006   -3.92e-06   -6.92e-07
ホテル客室数         0.0008      0.000      5.781      0.000       0.001       0.001
旅館客室数      -3.578e-05      0.000     -0.242      0.810      -0.000       0.000
==============================================================================
Omnibus:                       14.240   Durbin-Watson:                   1.048
Prob(Omnibus):                  0.001   Jarque-Bera (JB):               16.469
Skew:                           1.102   Prob(JB):                     0.000265
Kurtosis:                       4.885   Cond. No.                     6.05e+18
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The smallest eigenvalue is 1.91e-23. This might indicate that there are
strong multicollinearity problems or that the design matrix is singular.
"""

## 決定係数 

In [12]:
# 決定係数の表示
analysis.rsquared

0.5948301261418373

In [17]:
# 自由度調整済み決定係数の表示
analysis.rsquared_adj

0.566562460523826

決定係数は **0.5948301261418373** であり、モデルによって目的変数の **＊＊＊** %が説明できる事が分かった。また、自由度調整済み決定係数は **0.566562460523826** であった

## F検定

In [6]:
analysis.f_pvalue

1.5274367685392528e-08

有意水準を **0.05** としてモデル適合度のF検定を行った結果p値は **1.5e-0.8** であり、全ての偏回帰係数が0であるという帰無仮説が **棄却された** 。すなわち、今回の分析の重回帰モデルは **有意に妥当な変数が含まれている** 事が分かった。

# 2.偏回帰係数の解釈

In [20]:
analysis.pvalues
# 旅客客室のp値が0.05を超えるため、信頼するのが難しい(検定の観点だと相関がない。)

const     3.529716e-11
都道府県      3.529716e-11
人口        6.130946e-03
ホテル客室数    7.617077e-07
旅館客室数     8.098594e-01
dtype: float64

In [21]:
analysis.params

const     1.864018e+01
都道府県     -2.827862e-09
人口       -2.303980e-06
ホテル客室数    8.240426e-04
旅館客室数    -3.577930e-05
dtype: float64

> 有意水準を下回った変数の内、偏回帰係数の付合が負となったのは「人口」であり、偏回帰係数の付合が負となったのは「「ホテル客室数」であった。
> 変数「人口」の偏回帰係数は「x」であり、変数「人口」が1単位増加すると、目的変数「魅力度」は「x」減少する。
> 変数＊＊＊の偏回帰係数は＊＊＊であり、変数＊＊＊が1単位増加すると、目的変数＊＊＊は＊＊＊＜増加/減少＞する。